internal/controller/runtime/fsm/runtime_fsm_waiting_shoot_creation.go

```diff
@@ -31,6 +31,16 @@ func sFnWaitForShootCreation(_ context.Context, m *fsm, s *systemState) (stateFn
 
 	switch s.shoot.Status.LastOperation.State {
 	case gardener.LastOperationStateProcessing, gardener.LastOperationStatePending, gardener.LastOperationStateAborted, gardener.LastOperationStateError:
+		if stateNoMatchingSeeds(s.shoot) {
+			m.log.Info(fmt.Sprintf("Shoot %s has no matching seeds, setting error state", s.shoot.Name))
+			s.instance.UpdateStatePending(
+				imv1.ConditionTypeRuntimeProvisioned,
+				imv1.ConditionReasonCreationError,
+				"False",
+				"Shoot creation failed, no matching seeds")
+			return updateStatusAndRequeueAfter(m.RCCfg.RequeueDurationShootCreate)
```

Comment:
From issue description: 

 KIM has to stop the reconciliation and mark the RuntimeCR to be in ERROR state (non-matching seed **is a non-recoverable** error).


```suggestion
			return updateStatusAndStop()
```